### PR TITLE
CLN: Hashtable rename struct members

### PR DESCRIPTION
**Pull request by WillAyd**
Should help with readability

### CODE REVIEW

**Review by datapythonista — APPROVED** [review]
Very nice, thank you!

**Comment by datapythonista** [comment]
Seems like there is a conflict now, if you want to have a look...

**Comment by mroeschke** [comment]
One more conflict here otherwise LGTM

File: pandas/_libs/hashtable_class_helper.pxi.in [hunk 1]
```diff
@@ -164,7 +164,7 @@ ctypedef fused vector_data:
     StringVectorData
 
 cdef bint needs_resize(vector_data *data) noexcept nogil:
-    return data.n == data.m
+    return data.size == data.capacity
```
Comment (mroeschke) [inline pandas/_libs/hashtable_class_helper.pxi.in:167]: I don't know if `size` ever increments more than 1 before checking, but a `>=` would probably be safer here.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @WillAyd

**Pull request merged**